UEFITool/uefitool.cpp

```diff
@@ -647,6 +647,38 @@ void UEFITool::openImageFile(QString path)
 
     // Parse the image
     USTATUS result = ffsParser->parse(buffer);
+
+    // Show ffsParser's messages
+    std::vector<std::pair<UString, UModelIndex> > messages = ffsParser->getMessages();
+    for (size_t i = 0; i < messages.size(); i++) {
+        printf("%s\n", (const char *)messages[i].first.toLocal8Bit());
+    }
+
+    // Get last VTF
+    std::vector<std::pair<std::vector<UString>, UModelIndex > > fitTable = ffsParser->getFitTable();
+    if (fitTable.size()) {
+        printf("%s\n", "---------------------------------------------------------------------------");
+        printf("%s\n", "     Address      |   Size    |  Ver  | CS  |          Type / Info          ");
+        printf("%s\n", "---------------------------------------------------------------------------");
+        for (size_t i = 0; i < fitTable.size(); i++) {
+            printf("%s%s%s%s%s%s%s%s%s%s%s\n", (const char*)fitTable[i].first[0].toLocal8Bit(), " | "
```

Comment:
Feels really weird to me to add `, " | "` through an argument. Perhaps make it part of the formatted string?

Reply:
I wanted UEFITool to output to stdout what UEFIExtract does so that their execution/outputs can be compared more easily. But UEFIExtract uses iostream (cout::stdout) while UEFITool just uses printf. I forget if I tried iostream in UEFITool. Anyway, the printf calls look like that so that they look like the cout calls used by uefiextract_main.cp .

Is it ok to use iostream in UEFITool? If so, then I could do that and maybe make a common function that both UEFITool and UEFIExtract can use to output this info.

If we can't use iostream in UEFITool, then an alternative would be to replace cout with printf in UEFIExtract. That way I can still make both use a common function.

Reply:
I do not mind iostream in both, I think we would agree with @NikolajSchlej that both interfaces (iostream and printf) are not great in C++.